mungegithub/mungers/check-labels.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"k8s.io/contrib/mungegithub/features"
+	githubhelper "k8s.io/contrib/mungegithub/github"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/util/yaml"
+
+	"crypto/sha1"
+	yamlhelper "github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+type labelAccessor interface {
+	AddLabel(label *github.Label) error
+	GetLabels() ([]*github.Label, error)
+}
+
+// CheckLabelsMunger will check that the labels specified in the labels yaml file
+// are created.
+type CheckLabelsMunger struct {
+	labelFilePath string
+	prevSha       string
+	labelAccessor labelAccessor
+	features      *features.Features
+	reader        func() (string, error)
+}
+
+func init() {
+	RegisterMungerOrDie(&CheckLabelsMunger{})
+}
+
+// Name is the name usable in --pr-mungers
+func (c *CheckLabelsMunger) Name() string { return "check-labels" }
+
+// RequiredFeatures is a slice of 'features' that must be provided.
+func (c *CheckLabelsMunger) RequiredFeatures() []string { return []string{features.RepoFeatureName} }
+
+// Initialize will initialize the munger.
+func (c *CheckLabelsMunger) Initialize(config *githubhelper.Config, features *features.Features) error {
+	if len(c.labelFilePath) == 0 {
+		glog.Fatalf("No --label-file= supplied, cannot check labels")
+	}
+	c.labelAccessor = config
+	c.features = features
+	c.reader = func() (string, error) {
+		return c.features.Repos.ReadFile(c.labelFilePath)
+	}
+
+	if _, err := os.Stat(c.labelFilePath); os.IsNotExist(err) {
+		return fmt.Errorf("Failed to stat the check label config: %v", err)
+	}
+
+	return nil
+}
+
+func (c *CheckLabelsMunger) getSha(fileContents string) string {
+	h := sha1.New()
+	h.Write([]byte(fileContents))
```

Comment:
Consider canonicalizing the file before you hash it, so you don't trigger on e.g. sorts or whitespace changes